examples/kubernetes/kubernetes-config/main.tf

```diff
@@ -0,0 +1,147 @@
+terraform {
+  required_providers {
+    digitalocean = {
+      source  = "digitalocean/digitalocean"
+      version = ">= 2.4.0"
+    }
+    kubernetes = {
+      source = "hashicorp/kubernetes"
+      version = ">= 2.0.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">= 2.0.1"
+    }
+  }
+}
+
+data "digitalocean_kubernetes_cluster" "primary" {
+  name = var.cluster_name
+}
+
+resource "local_file" "kubeconfig" {
+  depends_on = [var.cluster_id]
+  count      = var.write_kubeconfig ? 1 : 0
+  content    = data.digitalocean_kubernetes_cluster.primary.kube_config[0].raw_config
+  filename   = "${path.root}/kubeconfig"
+}
+
+provider "kubernetes" {
+  host             = data.digitalocean_kubernetes_cluster.primary.endpoint
+  token            = data.digitalocean_kubernetes_cluster.primary.kube_config[0].token
+  cluster_ca_certificate = base64decode(
+    data.digitalocean_kubernetes_cluster.primary.kube_config[0].cluster_ca_certificate
+  )
```

Comment:
What ensures that the `data.digitalocean_kubernetes_cluster` resource is created after the actual `digitalocean_kubernetes_cluster` resource, i.e. what ensures that it contains the actual credentials and is not prone to ordering problems?

Reply:
Data sources always call out to the API, while a resource may use cached information. So stale credentials should not be returned.